### PR TITLE
DOM API: make compatible with Karax's requirements

### PR DESCRIPTION
**Pull request by Araq**

### CODE REVIEW

File: lib/js/dom.nim [hunk 1]
```diff
@@ -596,8 +1204,29 @@ proc decodeURIComponent*(uri: cstring): cstring {.importc, nodecl.}
...
+    top*, bottom*, left*, right*, x*, y*, width*, height*: float
+
+proc getBoundingClientRect*(e: Node): BoundingRect {.
+  importcpp: "getBoundingClientRect", nodecl.}
+proc clientHeight*(): int {.
+  importcpp: "(window.innerHeight || document.documentElement.clientHeight)@", nodecl.}
+proc clientWidth*(): int {.
+  importcpp: "(window.innerWidth || document.documentElement.clientWidth)@", nodecl.}
+
+proc inViewport*(el: Node): bool =
```
Comment (dom96) [inline lib/js/dom.nim]: This seems a little outside the scope of `dom`. Same for `clientWidth`/height.

File: lib/js/dom.nim [hunk 2]
```diff
@@ -405,62 +930,120 @@ type
...
+proc class*(n: Node): cstring {.importcpp: "#.className", nodecl.}
+proc `class=`*(n: Node; v: cstring) {.importcpp: "#.className = #", nodecl.}
+
+proc value*(n: Node): cstring {.importcpp: "#.value", nodecl.}
+proc `value=`*(n: Node; v: cstring) {.importcpp: "#.value = #", nodecl.}
+
+proc `disabled=`*(n: Node; v: bool) {.importcpp: "#.disabled = #", nodecl.}
+
+when defined(nodejs):
+  # we provide a dummy DOM for nodejs for testing purposes
```
Comment (dom96) [inline lib/js/dom.nim]: cool

File: lib/js/dom.nim (hunk 1 above)
Comment (Araq) [reply lib/js/dom.nim]: Karax requires it though, so I think this should stay. It also seems generally useful.

Comment (dom96) [reply lib/js/dom.nim]: Put it in Karax then?

**Pull request merged**